tests/e2e/concurrent-e2e/concurrent_test.go

```diff
@@ -0,0 +1,394 @@
+		pipelineSpec         numaflowv1.PipelineSpec
+		isbServiceSpec       numaflowv1.InterStepBufferServiceSpec
+		monovertexSpec       numaflowv1.MonoVertexSpec
+		pipelineVerifyFunc   func(numaflowv1.PipelineSpec) bool
+		isbServiceVerifyFunc func(numaflowv1.InterStepBufferServiceSpec) bool
+		monoVertexVerifyFunc func(numaflowv1.MonoVertexSpec) bool
+	}{
+		{
+			combination:          "Data Loss: Pipeline, ISBService, NumaflowController",
+			pipelineSpec:         dataLossPipelineSpec,
+			isbServiceSpec:       dataLossISBServiceSpec,
+			monovertexSpec:       updatedMonoVertexSpec,
+			pipelineVerifyFunc:   pipelineDataLossFunc,
+			isbServiceVerifyFunc: isbServiceDataLossFunc,
+			monoVertexVerifyFunc: monoVertexImageFunc,
+		},
+		{
+			combination:          "Data Loss: Pipeline, NumaflowController; Recreate: ISBService",
+			pipelineSpec:         dataLossPipelineSpec,
+			isbServiceSpec:       recreateFieldISBServiceSpec,
+			monovertexSpec:       updatedMonoVertexSpec,
+			pipelineVerifyFunc:   pipelineDataLossFunc,
+			isbServiceVerifyFunc: isbServiceRecreateFunc,
+			monoVertexVerifyFunc: monoVertexImageFunc,
+		},
+		{
+			combination:          "Data Loss: Pipeline, NumaflowController; Direct Apply: ISBService",
+			pipelineSpec:         dataLossPipelineSpec,
+			isbServiceSpec:       directApplyISBServiceSpec,
+			monovertexSpec:       updatedMonoVertexSpec,
+			pipelineVerifyFunc:   pipelineDataLossFunc,
+			isbServiceVerifyFunc: isbServiceDirectApplyFunc,
+			monoVertexVerifyFunc: monoVertexImageFunc,
+		},
+	}
+
+	for _, tc := range testCases {
+
+		It(fmt.Sprintf("Should update all rollouts concurrently\nCOMBINATION: %s", tc.combination), func() {
+
+			// create initial objects
+			CreateNumaflowControllerRollout(InitialNumaflowControllerVersion)
+			CreateISBServiceRollout(isbServiceRolloutName, initialISBServiceSpec)
+			CreatePipelineRollout(pipelineRolloutName, Namespace, initialPipelineSpec, false)
+			CreateMonoVertexRollout(monoVertexRolloutName, Namespace, initialMonoVertexSpec, nil)
+
+			// update each rollout
+			By("Updating PipelineRollout")
+			rawSpec, err := json.Marshal(tc.pipelineSpec)
+			Expect(err).ShouldNot(HaveOccurred())
+			UpdatePipelineRolloutInK8S(Namespace, pipelineRolloutName, func(rollout apiv1.PipelineRollout) (apiv1.PipelineRollout, error) {
+				rollout.Spec.Strategy = &apiv1.PipelineTypeRolloutStrategy{
+					PipelineTypeProgressiveStrategy: apiv1.PipelineTypeProgressiveStrategy{
+						Progressive: apiv1.ProgressiveStrategy{
+							AssessmentSchedule: "240,60,10",
+						},
+					},
+				}
+				rollout.Spec.Pipeline.Spec.Raw = rawSpec
+				return rollout, nil
+			})
+
+			By("Updating MonoVertexRollout")
+			rawSpec, err = json.Marshal(tc.monovertexSpec)
+			Expect(err).ShouldNot(HaveOccurred())
+			UpdateMonoVertexRolloutInK8S(monoVertexRolloutName, func(rollout apiv1.MonoVertexRollout) (apiv1.MonoVertexRollout, error) {
+				rollout.Spec.MonoVertex.Spec.Raw = rawSpec
```

Comment:
I'm thinking maybe we do 4 minutes for this one as well just because the Numaflow Controller is restarting. (and add the same comment)

It shouldn't create any extra delay since it's happening at the same time as the pipeline